qemu/qemu.go

```diff
@@ -646,6 +647,400 @@ const (
+
+	// Secs are the sectors
+	// Value: 1 <= Secs <= HardDiskSectorsMax
+	Secs uint8
+
+	// Trans forces the BIOS translation mode, this option is optional
+	// Values: none, lba, auto
+	Trans HardDiskTranslationMode
+}
+
+// HardDiskTranslationMode hard disk translation modes
+type HardDiskTranslationMode string
+
+const (
+	// HardDiskTranslationAuto auto translation mode
+	HardDiskTranslationAuto HardDiskTranslationMode = "auto"
+
+	// HardDiskTranslationNone none translation mode
+	HardDiskTranslationNone HardDiskTranslationMode = "none"
+
+	// HardDiskTranslationLba lba translation mode
+	HardDiskTranslationLba HardDiskTranslationMode = "lba"
+)
+
+const (
+	// HardDiskCylindersMax max number of cylinders
+	HardDiskCylindersMax uint16 = 16383
+
+	// HardDiskHeadsMax max number of heads
+	HardDiskHeadsMax uint8 = 16
+
+	// HardDiskSectorsMax max number of sectors
+	HardDiskSectorsMax uint8 = 63
+)
+
+// Valid returns true if the hard disk geometry is valid, else false
+func (h *HardDiskGeometry) Valid() bool {
+	if h.Cyls < 1 || h.Cyls > HardDiskCylindersMax {
+		return false
+	}
+
+	if h.Heads < 1 || h.Heads > HardDiskHeadsMax {
+		return false
+	}
+
+	if h.Secs < 1 || h.Secs > HardDiskSectorsMax {
+		return false
+	}
+
+	if h.Trans != "" {
+		switch h.Trans {
+		case HardDiskTranslationAuto, HardDiskTranslationLba, HardDiskTranslationNone:
+			break
+		default:
+			return false
+		}
+	}
+
+	return true
+}
+
+// DriveConnection defines where is connected the drive
+type DriveConnection struct {
+	// Bus where the drive is connected
+	Bus string
+
+	// Unit id where the drive is connected
+	Unit uint
+}
+
+// DriveRWErrors specifies which action to take on write and read errors
+type DriveRWErrors struct {
+	// Werror specifies which action to take on write errors
+	// Available values are: ignore, stop, report, enospc
+	Werror DriveRWErrorsAction
+
+	// Rerror specifies which action to take on read errors
+	// Available values are: ignore, stop, report, enospc
+	Rerror DriveRWErrorsAction
+}
+
+// DriveRWErrorsAction is action to take on rw errors
+type DriveRWErrorsAction string
+
+const (
+	// DriveRWErrorsIgnore ignores the error
+	DriveRWErrorsIgnore DriveRWErrorsAction = "ignore"
+
+	// DriveRWErrorsStop stops Qemu
+	DriveRWErrorsStop DriveRWErrorsAction = "stop"
+
+	// DriveRWErrorsReport reports the error
+	DriveRWErrorsReport DriveRWErrorsAction = "report"
+
+	// DriveRWErrorsEnospc pauses Qemu only if the host disk is full
+	DriveRWErrorsEnospc DriveRWErrorsAction = "enospc"
+)
+
+// Valid returns true if the drive rw error is valid, else false
+func (d *DriveRWErrors) Valid() bool {
+	switch d.Werror {
+	case DriveRWErrorsIgnore, DriveRWErrorsStop, DriveRWErrorsReport, DriveRWErrorsEnospc:
+		break
+	default:
+		return false
+	}
+
+	switch d.Rerror {
+	case DriveRWErrorsIgnore, DriveRWErrorsStop, DriveRWErrorsReport, DriveRWErrorsEnospc:
+		break
+	default:
+		return false
+	}
+
+	return true
+}
+
+// Drive represents a qemu drive.
```

Comment:
The other comments on the structures you've added are not terminated by a full stop.

Reply:
fixed